src/Magnum/Platform/WindowlessEglApplication.cpp

```diff
@@ -84,6 +84,23 @@ bool extensionSupported(const char* const extensions, Containers::ArrayView<cons
 WindowlessEglContext::WindowlessEglContext(const Configuration& configuration, GLContext* const magnumContext) {
     #ifndef MAGNUM_TARGET_WEBGL
     
+	/**
+	 * The user provided a shared context and its associated display, so we don't query the display
+	 */
+   	if(configuration.sharedContext() != EGL_NO_CONTEXT &&
+   	  configuration.sharedDisplay() != EGL_NO_DISPLAY)
+   	{   	      		
+   		_display = configuration.sharedDisplay();
+   	    _sharedContext = true; 
+   	}
+   	/**
+     * The user provided only the shared context or display, or provided both but one of them is wrong
+     */
+    else if (configuration.sharedContext() == EGL_NO_CONTEXT && configuration.sharedDisplay() != EGL_NO_DISPLAY ||
+             configuration.sharedContext() != EGL_NO_CONTEXT && configuration.sharedDisplay() == EGL_NO_DISPLAY)
+   {
+     	Error{} << "Platform::WindowlessEglContext: Either context or display is not valid. Context is " << configuration.sharedContext() << " Display is " << configuration.sharedDisplay();  	    
+   }   
```

Comment:
Is it just me or is the indentation drunk here? :D Same below.

Reply:
Corrected !